### PR TITLE
replace json theia extension with vscode one

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

Fixes: #7981 
Fixes: #3983 

- [x] fix 7981, fix 3983: replace json theia extension with vscode one
- [x] document breaking changes
- [x] deprecate contributing language support from Theia extension

Please see [CHANGELOG](https://github.com/eclipse-theia/theia/blob/ak/json_vscode/CHANGELOG.md#1_4_0_replace_json) for breaking changes.

#### How to test

- verify that smartness available in settings (user, workspace), debug and task files as before
- verify that user data as settings and keymaps are preserved
- verify that jsonValidation contribution point from VS Code extension is respected
- verify `How to test` from https://github.com/eclipse-theia/theia/pull/7225

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/json-schema-store.ts [hunk 1]
```diff
@@ -14,61 +14,40 @@
...
-                }));
-            }
+    onStart(): void {
+        for (const contribution of this.contributions.getContributions()) {
+            contribution.registerSchemas(this);
         }
+    }
+
+    /**
+     * Clients should implement `JsonSchemaContribution` to call these method,
```
Comment (akosyakov) [inline packages/core/src/browser/json-schema-store.ts:46]: these -> this

File: packages/json/src/browser/json-client-contribution.ts [hunk 2]
```diff
@@ -1,131 +0,0 @@
...
-            if (s.fileMatch) {
-                this.jsonSchemaStore.registerSchema({
-                    fileMatch: s.fileMatch,
-                    url: s.url
-                });
-            }
-        }
-    }
-
-}
```
Comment (kittaakos) [inline packages/json/src/browser/json-client-contribution.ts:122]: We have to restore this.

Comment (akosyakov) [reply packages/json/src/browser/json-client-contribution.ts:122]: it should be resolved now

**Comment by akosyakov** [comment]
@vince-fugnitto How do you reproduce it. I've just tried latest state in Gitpod:
settings

Could you check terminal that VS Code extensions are downloaded successfully for you?

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto How do you reproduce it. I've just tried latest state in Gitpod:

I re-verified and it works correctly now, I'm not sure how I got in that weird state before.

File: packages/core/src/browser/json-schema-store.ts [hunk 3]
```diff
@@ -14,65 +14,111 @@
...
+            this._schemas.resolve(schemas);
+        };
+        const registerTimeout = this.getRegisterTimeout();
+        for (const contribution of this.contributions.getContributions()) {
+            let warned = false;
+            const result = contribution.registerSchemas({
+                registerSchema: schema => {
+                    if (sealed) {
+                        if (!warned) {
+                            console.warn(`${contribution.constructor.name}.registerSchemas is taking more than (${registerTimeout.toFixed(1)} ms), new schemas are ignored.`);
```
Comment (akosyakov) [inline packages/core/src/browser/json-schema-store.ts:64]: it is better to throw here.

**Comment by akosyakov** [comment]
It is required for https://github.com/eclipse-theia/theia/issues/7100

@vince-fugnitto Could you have a look again please? Is there anything else should be addressed?

**Comment by bd82** [comment]
Hello @akosyakov 

What is the behavior when multiple jsonSchemas are registered for an identical fileMatch pattern?
For example: If there is a schemaStore registration for `xyz.json` files and also a VSCode Ext with `jsonContribution` forh `xyz.json`?

Basically do `jsonContribution` **replace** schemaStore registrations or are they **added as extra** schemas?

File: packages/task/src/browser/task-schema-updater.ts [hunk 4]
```diff
@@ -55,13 +53,17 @@ export class TaskSchemaUpdater {
...
-            if (uri.toString() === taskSchemaUri.toString()) {
+        const resource = this.inmemoryResources.add(taskSchemaUri, '');
+        if (resource.onDidChangeContents) {
+            resource.onDidChangeContents(() => {
                 this.onDidChangeTaskSchemaEmitter.fire(undefined);
-            }
+            });
+        }
+        context.registerSchema({
+            fileMatch: ['tasks.json', USER_TASKS_URI.toString()],
```
Comment (azatsarynnyy) [inline packages/task/src/browser/task-schema-updater.ts:65]: @akosyakov when I'm trying to configure a global task, I get:
```
resource.ts:166 Uncaught (in promise) Error: A resource provider for 'user_storage:tasks.json' is not registered.
    at DefaultResourceProvider.<anonymous> (resource.ts:166)
    at step (resource.ts:15)
    at Object.throw (resource.ts:15)
    at rejected (resource.ts:15)
```

**Comment by akosyakov** [comment]
> Basically do jsonContribution replace schemaStore registrations or are they added as extra schemas?

They are added as extra schemas. You will need to check how json built-in extension handles it.

File: packages/task/src/browser/task-schema-updater.ts (hunk 4 above)
Comment (akosyakov) [reply packages/task/src/browser/task-schema-updater.ts:65]: @azatsarynnyy good catch! could you try again please?

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,11 +6,15 @@
...
   - You can register JSON validations at application startup by implementing `JsonSchemaContribution` Theia contribution point.
   - Alternatively you can provide JSON validations using VS Code [contributes.jsonValidation](https://code.visualstudio.com/api/references/contribution-points#contributes.jsonValidation) contribution point.
-- <a name="1_4_0_absolute_user_storage_uri">[user-storage]</a> settings URI must be an absolute to satisfy expectations of `vscode.json-language-features` [#7972](https://github.com/eclipse-theia/theia/pull/7972)
+<a name="1_4_0_absolute_user_storage_uri"></a>
+- [[user-storage]](#1_4_0_absolute_user_storage_uri) settings URI must be an absolute to satisfy expectations of `vscode.json-language-features` [#8112](https://github.com/eclipse-theia/theia/pull/8112)
   - If you imlpement a custom user storage make sure to check old relaitve locations, otherwise it can cause user data loss.
+<a name="1_4_0_deprecate_languages"></a>
+- [[languages]](#1_4_0_deprecate_languages) `@theia/languages` extension is deprecated, use VS Code extensions to provide language smartness [#8112](https://github.com/eclipse-theia/theia/pull/8112)
```
Comment (akosyakov) [inline CHANGELOG.md:17]: point to VS Code docs about how

File: packages/task/src/browser/task-schema-updater.ts (hunk 4 above)
Comment (azatsarynnyy) [reply packages/task/src/browser/task-schema-updater.ts:65]: @akosyakov it's fixed now. Thanks!

**Comment by akosyakov** [comment]
I'm updating deprecation notes with a link to VS Code docs about how to integrate a language server via VS Code extension and merging if there are no objections.

**Pull request merged**